AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.java

```diff
@@ -223,7 +223,16 @@ public MaterialDialog onCreateDialog(Bundle savedInstanceState) {
                                         return true;
                                     });
                 }
-                return builder.show();
+                MaterialDialog materialDialog = builder.build();
+                materialDialog.setOnKeyListener((dialog, keyCode, event) -> {
+                    if (keyCode == KeyEvent.KEYCODE_BACK) {
+                        Timber.i("catch");
+                        dismissAllDialogFragments();
+                        return true;
+                    }
+                    return false;
+                });
+                return materialDialog;
```

Comment:
It is the responsibility of each of the dialogs to actually show the dialog as well I think? I think you have to do that call? I'm not sure how this would have tested correctly without it but maybe I am missing something so this suggestion may not be correct

```suggestion
                });
                materialDialog.show();
                return materialDialog;
```

Reply:
This is in an `AsyncDialogFragment`: https://developer.android.com/reference/android/app/DialogFragment#onCreateDialog(android.os.Bundle)

Doesn't seem to specify that you need to show the returned dialog, many examples don't, and `show()` is called during `onStart`

Reply:
Okay - I was basing more off all the other cases showing. Seemed it should be conformant based on other usage locally to this code